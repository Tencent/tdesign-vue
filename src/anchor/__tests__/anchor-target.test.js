import { mount } from '@vue/test-utils';

jest.resetModules();
let text = '';
jest.mock('@/src/utils/clipboard', () => (arg) => (text = arg));

// every component needs four parts: props/events/slots/functions.
describe('AnchorTarget', () => {
  const { AnchorTarget } = require('@/src/anchor/index');
  // test props api
  describe('test copy', () => {
    const wrapper = mount(AnchorTarget, {
      propsData: {
        id: 'test-target',
      },
    });

    const a = wrapper.find('.t-icon-file-copy');
    a.trigger('click');
    expect(text).toEqual(`${location.href}#test-target`);
  });

  // test slots
  describe('<slot>', () => {
    it('should render default slot', async () => {
      const wrapper = mount(AnchorTarget, {
        propsData: {
          id: 'test-target',
          tag: 'h1',
        },
        slots: {
          default: 'this is default slots',
        },
      });
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
