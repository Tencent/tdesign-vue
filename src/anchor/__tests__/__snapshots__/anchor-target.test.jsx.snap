// Vitest Snapshot v1

exports[`AnchorTarget > <slot> > should render default slot 1`] = `
"<h1 id=\\"test-target\\" class=\\"t-anchor__target\\">this is default slots<svg fill=\\"none\\" viewBox=\\"0 0 24 24\\" width=\\"1em\\" height=\\"1em\\" class=\\"t-icon t-icon-file-copy t-anchor__copy\\">
    <path fill=\\"currentColor\\" d=\\"M6 1H15.4142L21 6.58579V19H6V1ZM8 3V17H19V9H13V3H8ZM15 3.41421V7H18.5858L15 3.41421ZM4 5V21H15V23H2V5H4Z\\"></path>
  </svg></h1>"
`;
